contracts/smart-account/modules/BatchedSessionRouterModule.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {BaseAuthorizationModule, UserOperation, ISignatureValidator} from "./BaseAuthorizationModule.sol";
+import {ISessionValidationModule} from "./SessionValidationModules/ISessionValidationModule.sol";
+import {ISessionKeyManager} from "../interfaces/ISessionKeyManager.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@account-abstraction/contracts/core/Helpers.sol";
+
+/**
+ * @title Batched Session Router
+ * @dev Built to process executeBatch and executeBatch_y6U calls
+ *         - Every call inside batch should be covered by an appropriate Session Validation Module
+ *         - Parses data provided and sequentially 
+                a) verifies the session key was enabled via SessionKeyManager
+                b) verifies the session key permissions via Session Validation Modules
+ *         - Should be used with carefully verified and audited Session Validation Modules only
+ *         - Compatible with Biconomy Modular Interface v 0.1
+ * @author Fil Makarov - <filipp.makarov@biconomy.io>
+ */
+
+contract BatchedSessionRouter is BaseAuthorizationModule {
+    bytes4 public constant EXECUTE_BATCH_SELECTOR = 0x47e1da2a;
+    bytes4 public constant EXECUTE_BATCH_OPTIMIZED_SELECTOR = 0x00004680;
+
+    /**
+     * @dev validates userOperation. Expects it to be a executeBatch or executeBatch_y6U call
+     * If something goes wrong, reverts
+     * @param userOp User Operation to be validated.
+     * @param userOpHash Hash of the User Operation to be validated.
+     * @return SIG_VALIDATION_FAILED or packed validation result.
+     */
+    function validateUserOp(
+        UserOperation calldata userOp,
+        bytes32 userOpHash
+    ) external virtual returns (uint256) {
+        // check this is a proper method call
+        require(
+            bytes4(userOp.callData[0:4]) == EXECUTE_BATCH_OPTIMIZED_SELECTOR ||
+                bytes4(userOp.callData[0:4]) == EXECUTE_BATCH_SELECTOR,
+            "SR Invalid Selector"
+        );
+
+        (bytes memory moduleSignature, ) = abi.decode(
+            userOp.signature,
+            (bytes, address)
+        );
+
+        // parse the signature to get the array of required parameters
+        (
+            address sessionKeyManager,
+            uint48[] memory validUntil,
+            uint48[] memory validAfter,
+            address[] memory sessionValidationModule,
+            bytes[] memory sessionKeyData,
+            bytes32[][] memory merkleProof,
+            bytes memory sessionKeySignature
+        ) = abi.decode(
+                moduleSignature,
+                (
+                    address,
+                    uint48[],
+                    uint48[],
+                    address[],
+                    bytes[],
+                    bytes32[][],
+                    bytes
+                )
+            );
+
+        // check lengths of arrays
+        require(
```

Comment:
Multiple require statement can be replaced by single require statement to save gas

```
require(
    validUntil.length == validAfter.length &&
    validUntil.length == sessionValidationModule.length &&
    validUntil.length == sessionKeyData.length &&
    validUntil.length == merkleProof.length,
    "SR Invalid data provided"
);
```

Reply:
actually multiple require statements are cheaper
https://github.com/code-423n4/2023-01-biconomy-findings/blob/main/data/Aymen0909-G.md#5--splitting-require-statements-that-uses--saves-gas-

Reply:
In any case, I removed this competely